libs/application/ui-shell/src/lib/ApplicationForm.tsx

```diff
@@ -42,11 +49,18 @@ const ApplicationLoader: FC<{
   if (loading) {
     return null
   }
+
   return (
-    <ShellWrapper
-      application={application}
-      nationalRegistryId={nationalRegistryId}
-    />
+    <RefetchProvider
+      value={() => {
+        refetch()
+      }}
+    >
+      <ShellWrapper
+        application={application}
```

Comment:
Since we have access to the application object, I'm wondering if a state field in the object would make more sense. Then we can listen if the state is in draft/review and refetch accordingly when needed.

Using a provider in this case looks a bit off in my opinion, but I might be wrong

Reply:
Will leave this as is for now, until we can figure out a better approach. 
Related: https://github.com/apollographql/apollo-client/issues/3633
We run into the same (non ui-update issue) when trying to use `refetchQueries`